cpp/include/cudf/unary.hpp

```diff
@@ -31,6 +33,65 @@ namespace cudf {
  * @brief Column APIs for unary ops
  */
 
+/**
+ * @brief Convert a floating-point value to fixed-point
+ *
+ * @tparam Fixed The fixed-point type we are converting to
+ * @tparam Floating The floating-point type we are converting from
+ * @param floating The floating-point value to convert
+ * @param scale The desired scale of the fixed-point value
+ * @return The converted to fixed-point value
+ */
+template <typename Fixed,
+          typename Floating,
+          typename cuda::std::enable_if_t<is_fixed_point<Fixed>() &&
+                                          cuda::std::is_floating_point_v<Floating>>* = nullptr>
+CUDF_HOST_DEVICE Fixed convert_floating_to_fixed(Floating floating, numeric::scale_type scale)
+{
+  using Rep          = typename Fixed::rep;
+  auto const shifted = numeric::detail::shift<Rep, Fixed::rad>(floating, scale);
+  numeric::scaled_integer<Rep> scaled{static_cast<Rep>(shifted), scale};
+  return Fixed(scaled);
+}
+
+/** @brief Convert a fixed-point value to floating-point
+ *
+ * @tparam Floating The floating-point type we are converting to
+ * @tparam Fixed The fixed-point type we are converting from
+ * @param fixed The fixed-point value to convert
+ * @return The converted-to floating-point value.
```

Comment:
Suggestion: only hyphenate "floating point" when it is used as a compound adjective.

```suggestion
/** @brief Convert a fixed-point value to floating point.
 *
 * @tparam Floating The floating-point type to convert to.
 * @tparam Fixed The fixed-point type to convert from.
 * @param fixed The fixed-point value to convert.
 * @return The converted floating-point value.
```